File: ColtSteele/capitalizeWords.py
"""
Given an array of words, return a new array containing each word capitalized
Input: array
Output: array
"""

def capitalizeWords(arr):

    if len(arr) == 0:
        return []

    return [arr[-1].upper()] + capitalizeWords(arr[:-1])

if __name__ == "__main__":
    print(capitalizeWords(["In", "computer", "science", 
    "tree", "traversal", "is", "a", "form", "of", "graph"]))

File: ColtSteele/capitalizeWords.js
/**
 * Given an array of words, return a new array containing each word capitalized
 */
function capitalizeWords(array) {
  if (array.length === 1) {
    return [array[0].toUpperCase()];
  }
  let res = capitalizeWords(array.slice(0, -1));
  res.push(array.slice(array.length - 1)[0].toUpperCase());
  return res;
}

console.log(capitalizeWords(['i', 'am', 'learning', 'recursion']));